### DISCUSSION TITLE
Filtering by a dynamic formula field

### DISCUSSION THREAD

**Discussion by Suki0011** [Budibase/budibase#10644] [Help] [answered]
Hi!

I have a table with a start date, a finish date and a formula field that calculates remaining days. 
I want to make a view that filters rows that are within 60 days of the finish date, but as you know you cant filter using dynamic formula fields, and the field cant be static since it has to calculate remaining days every time the form is opened.

Is there a way to copy a value from the formula field to a text field every time the table i refreshed, so that i could filter bi that text field, or is there any other way of doing this that hasnt occurred to me?

Thanks!

Suki

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Suki0011,

I think you can achieve something like this through automations. You will likely need to create 2 different automations, one for create and one for the update events.

Both of these automations will do the exact same thing however, they listen for different events.

Not sure what your dates between function looks like so I'll link the one i've written for the automations below:

```
const date1 = new Date($("trigger.row.Start Date"));
const date2 = new Date($("trigger.row.End Date"));

const diffInMs = Math.abs(date2 - date1);

const diffInDays = Math.ceil(diffInMs / (1000 * 60 * 60 * 24));
if (diffInDays < 1) {
	return 1
}
return diffInDays;
```

1. Initially I made a table called `Date Check` columns below. Ignore `Days between dates` this was the formula field used to test the code to get the dates between. Automation copy is what we will be using to get the days between dates as a number value.
<img width="1179" alt="Screenshot 2023-05-23 at 10 35 01" src="https://github.com/Budibase/budibase/assets/126772285/b24a098b-419b-4da5-b774-49fd1ed3029e">
2. Created two different automations one for your `create` and `update` events attached to the table you want it to listen on. Make sure the create automation uses `ROW CREATED` where the updated automations uses `ROW UPDATED`.
<img width="284" alt="Screenshot 2023-05-23 at 10 34 08" src="https://github.com/Budibase/budibase/assets/126772285/efb5c207-a727-42b3-9b62-6eef554bc77a">

**Example of the initial row created trigger attached to relevant table**
<img width="534" alt="Screenshot 2023-05-23 at 10 34 23" src="https://github.com/Budibase/budibase/assets/126772285/ef10f428-faef-4da3-930d-99879f79a71a">
3. Next add the step `Update Row`.
<img width="531" alt="Screenshot 2023-05-23 at 10 34 32" src="https://github.com/Budibase/budibase/assets/126772285/19511357-ac99-4519-b0f6-7edff689f08c">
4. Select `Use bindings` for the the update row step.
<img width="533" alt="Screenshot 2023-05-23 at 10 54 15" src="https://github.com/Budibase/budibase/assets/126772285/016e730b-7d26-4ece-bfe8-e29d24011917">
5. Fill in all relevant information on the update row step. Make sure to use the `trigger.id` as the row id.
<img width="532" alt="Screenshot 2023-05-23 at 10 34 40" src="https://github.com/Budibase/budibase/assets/126772285/b96f4724-51ff-43a4-8ddc-dc7fbed04f17">
6. Paste the function code shown above into the new automation copy field, alter this if your field has different names.

**The end result**
![Automation output days between](https://github.com/Budibase/budibase/assets/126772285/839a896d-925d-4947-a51e-49622c7a8381)

**Side note** you can achieve this using a single automation through using the `App action` automation type. The difference being you will need to pass in the relevant information through `Trigger Automation` on click action in the design area.  Lastly, ensure you publish your app as automations only run on apps which have been published.

I hope this helps you!

**Reply by Suki0011** [reply]
Thank you so much @ConorWebb96, I never thought of using automations in such a way.
